rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/ImageNameFactory.groovy

```diff
@@ -16,27 +16,73 @@
 
 package com.netflix.spinnaker.rosco.providers.util
 
+import com.netflix.frigga.ami.AppVersion
 import com.netflix.spinnaker.rosco.api.BakeOptions
 import com.netflix.spinnaker.rosco.api.BakeRequest
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
 
-interface ImageNameFactory {
+import java.time.Clock
 
   /**
-   * This method is responsible for:
-   *   - Producing an image name.
-   *   - (If one or more package names are specified) Attempting to derive the AppVersion descriptor from the
-   *     first package name. If the AppVersion could not be derived, this value will be null.
-   *
-   * This method always returns a list of size 3 with the following elements:
-   *   1) A derived image name (to be used for naming the image being baked).
-   *   2) The appversion string (to be used for tagging the newly-baked image).
-   *   3) The updated list of packages to be used for overriding the passed package list (this will be removed once the
-   *      temporary workaround described above is removed).
-   *
-   * This function is not required to return the same image name on multiple invocations with the same bake request.
-   *
-   * Returns [imageName, appVersionStr, packagesParameter].
+   * Default implementation of ImageNameFactory considers only package_name, a timestamp, and base_os.
+   * For more fine grained conventions, extend this class, override the required methods and alter your
+   * provider specific RoscoConfiguration to reflect the change.
    */
-  def deriveImageNameAndAppVersion(BakeRequest bakeRequest, BakeOptions.Selected selectedOptions)
+
+@Slf4j
+public class ImageNameFactory {
+
+  @Autowired
+  Clock clock
+
+  /**
+   * Attempts to produce an appVersionStr from the first packageName; to be used for tagging the newly-baked image
+   */
+  def buildAppVersionStr(BakeRequest bakeRequest, PackageNameConverter.OsPackageName osPackageName) {
+    String appVersionStr = PackageNameConverter.buildAppVersionStr(bakeRequest, osPackageName)
+
+    if (!AppVersion.parseName(appVersionStr)) {
+      if (appVersionStr) {
+        log.debug("AppVersion.parseName() was unable to parse appVersionStr=$appVersionStr.")
+      }
+      return null
+    }
+    return appVersionStr
+  }
+
+  /**
+   * Produces an imageName either from the BakeRequest.ami_name or the first package to be installed.
+   * This is to be used for naming the image being baked. Note that this function is not required to
+   * return the same image name on multiple invocations with the same bake request
+   */
+  def buildImageName(BakeRequest bakeRequest, List<PackageNameConverter.OsPackageName> osPackageNames) {
+    String timestamp = clock.millis()
+    def emptyBasePackage = new PackageNameConverter.OsPackageName()
```

Comment:
This feels a little like abusing the `OsPackageName` model, especially since the `buildOsPackageName()` factory method throws an exception if it's not `DEB` or `RPM`. Maybe add an explicit type there for this use case?
If you decide you really want to keep it this way, you should be able to use: `def emptyBasePackage = new PackageNameConverter.OsPackageName(name: '', arch: 'all')`